### PR TITLE
update builtins, cleanup pre-1.6 code paths

### PR DESCRIPTION
**Pull request by simeonschaub**
This makes `src/builtins.jl` in sync with `bin/generate_builtins.jl`
again. A lot of this code was also out-of-date, now that we only support
1.6, so this includes some cleanup.

### CODE REVIEW

**Review by aviatesk — APPROVED** [review]
LGTM. I'd like to ask a review from @KristofferC before merging, as I remember we decided to turn off the automated builtin file generation for good reasons (see https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/450).

Having said that, the discussion I liked above may be unrelated to this PR as this one doesn't enable the installation time generation.

**Comment by KristofferC** [comment]
> I'd like to ask a review from @KristofferC before merging, as I remember we decided to turn off the automated builtin file generation for good reasons (see https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/450).

We turned off running it during compile time and having one builtin file per version but just updating the generation file seems uncontroversial. So LGTM (in principle, didn't look through carefully the exact changes but if tests pass I'm sure it's good).

File: bin/generate_builtins.jl [hunk 1]
```diff
@@ -137,47 +138,79 @@ function maybe_evaluate_builtin(frame, call_expr, expand::Bool)
...
+        end
+        return maybe_recurse_expanded_builtin(frame, new_expr)
+""")
         end
 
         id = findfirst(isequal(f), Core.Compiler.T_FFUNC_KEY)
         fcall = generate_fcall(f, Core.Compiler.T_FFUNC_VAL, id)
-        print(io,
+        if f in Core.Builtin[
+            Core._call_in_world_total, Core.donotdelete,
```
Comment (KristofferC) [inline bin/generate_builtins.jl:197]: This requires Julia 1.9 to run the file now. Maybe that is ok.

Comment (simeonschaub) [reply bin/generate_builtins.jl:197]: Yeah, I don't think there really is a way around that if we want the list to be up-to-date. I can maybe add a comment to that regard though

**Comment by simeonschaub** [comment]
Seems like https://github.com/JuliaLang/julia/pull/44855 hasn't arrived on nightly yet, that might be part of the reason nightly CI is so slow currently

**Pull request merged**